### PR TITLE
Split up services.xml into multiple files

### LINKED ISSUE #150: Split up services.xml
Right now we load everything by default and then remove services in the bundle's extension. 

One could use multiple files for service declaration and then load ie `twig.xml` if the twig config is enabled. Any thoughts?

There is no real pros or cons, but it may make our extension class cleaner.

### PR DESCRIPTION
**Pull request by Nyholm**
This will fix #150.

### CODE REVIEW

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -54,12 +54,15 @@ public function getConfigTreeBuilder()
                 ->booleanNode('log_exceptions')
                     ->defaultFalse()
                 ->end()
-                ->booleanNode('log_deprecations')
-                    ->defaultFalse()
-                ->end()
                 ->booleanNode('log_commands')
```
Comment (jderusse) [inline DependencyInjection/Configuration.php]: what's about moving `ignored_commands` inside this section? Same optimizations for `http`

Something like this (can be improved)
```yaml
ekino_new_relic:
    application_name: Awesome Application
    deployment_names: ~

    api_key:
    license_key:

    xmit: false
    logging: false
    instrument: false

    log_exceptions: false

    log_deprecations: false

    log_logs:
       channels: [app]
       level: error
       service: app.my_custom_handler
    
    log_commands:
        ignored_commands: []
    
    http:
        using_symfony_cache: false
        transaction_naming: route
        transaction_naming_service: ~
        ignored_routes: []
        ignored_paths: []
    

    twig: true
```

File: DependencyInjection/Compiler/MonologHandlerPass.php [hunk 2]
```diff
@@ -10,16 +10,12 @@ class MonologHandlerPass implements CompilerPassInterface
 {
     public function process(ContainerBuilder $container)
     {
-        if (!$container->hasParameter('ekino.new_relic.log_logs') || !$container->hasDefinition('monolog.logger')) {
+        if (!$container->hasParameter('ekino.new_relic.monolog.channels') || !$container->hasDefinition('monolog.logger')) {
```
Comment (jderusse) [inline DependencyInjection/Compiler/MonologHandlerPass.php]: shouldn't we add a test to assert that, when config is disabled, the key channel does not exists?

File: DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -46,21 +45,62 @@ public function getConfigTreeBuilder()
...
+                            ->end()
+                        ->end()
+                        ->scalarNode('using_symfony_cache')->defaultFalse()->end()
+                        ->booleanNode('instrument')
+                            ->defaultFalse()
+                        ->end()
+                    ->end()
                 ->end()
-                ->arrayNode('log_logs')
+                ->arrayNode('monolog')
```
Comment (jderusse) [inline DependencyInjection/Configuration.php]: :1+:

File: DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -46,21 +45,62 @@ public function getConfigTreeBuilder()
...
                     ->defaultFalse()
                 ->end()
-                ->booleanNode('instrument')
-                    ->defaultFalse()
+                ->arrayNode('exceptions')
+                    ->canBeDisabled()
                 ->end()
-                ->booleanNode('log_exceptions')
-                    ->defaultFalse()
+                ->arrayNode('commands')
```
Comment (jderusse) [inline DependencyInjection/Configuration.php]: README should be updated

Comment (Nyholm) [reply DependencyInjection/Configuration.php]: Done

**Comment by Nyholm** [comment]
Thanks

**Pull request merged**